### PR TITLE
feat(api-keys): implement 'list api keys' function for query

### PR DESCRIPTION
**Pull request by vindard**

### CODE REVIEW

File: core/api-keys/src/identity/mod.rs [hunk 1]
```diff
@@ -105,4 +105,39 @@ impl Identities {
...
+        &self,
+        subject_id: &str,
+    ) -> Result<Vec<IdentityApiKey>, IdentityError> {
+        let identity = sqlx::query!(
+            r#"SELECT id FROM identities WHERE subject_id = $1"#,
+            subject_id,
+        )
+        .fetch_optional(&self.pool)
+        .await?
+        .ok_or(IdentityError::NoIdentityForSubject)?;
```
Comment (vindard) [inline core/api-keys/src/identity/mod.rs:119]: Should we create the identity here and return an empty array instead?

Comment (bodymindarts) [reply core/api-keys/src/identity/mod.rs:119]: not here but in app layer we should return an empty list

File: core/api-keys/src/identity/mod.rs [hunk 2]
```diff
@@ -105,4 +105,39 @@ impl Identities {
...
+
+        let api_keys_records = sqlx::query!(
+            r#"SELECT id, name, created_at, expires_at
```
Comment (bodymindarts) [inline core/api-keys/src/identity/mod.rs:122]: not 2 queries - use a JOIN

File: core/api-keys/src/identity/mod.rs [hunk 3]
```diff
@@ -105,4 +105,39 @@ impl Identities {
...
+               FROM identity_api_keys
+               WHERE identity_id = $1 AND active = true AND expires_at > NOW() AT TIME ZONE 'utc'"#,
```
Comment (bodymindarts) [inline core/api-keys/src/identity/mod.rs:124]: I think we can show `expired` keys - but not active ones. (active == revoked)

File: core/api-keys/src/identity/mod.rs (hunk 1 above)
Comment (vindard) [reply core/api-keys/src/identity/mod.rs]: Solved with join

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**